src/react-queries/queries/accountManagement.ts

```diff
@@ -2,67 +2,75 @@ import { createQueryKeys } from '@lukemorales/query-key-factory'
 import type { AccountId } from '@shapeshiftoss/caip'
 import { type ChainId, fromAccountId } from '@shapeshiftoss/caip'
 import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { AccountMetadata } from '@shapeshiftoss/types'
 import { deriveAccountIdsAndMetadata } from 'lib/account/account'
 import { assertGetChainAdapter } from 'lib/utils'
 import { checkAccountHasActivity } from 'state/slices/portfolioSlice/utils'
 
-const getAccountIdsWithActivity = async (
+const getAccountIdsWithActivityAndMetadata = async (
   accountNumber: number,
   chainId: ChainId,
   wallet: HDWallet,
 ) => {
   const input = { accountNumber, chainIds: [chainId], wallet }
   const accountIdsAndMetadata = await deriveAccountIdsAndMetadata(input)
-  const [accountId] = Object.keys(accountIdsAndMetadata)
+  const [[accountId, accountMetadata]] = Object.entries(accountIdsAndMetadata)
 
   const { account: pubkey } = fromAccountId(accountId)
   const adapter = assertGetChainAdapter(chainId)
   const account = await adapter.getAccount(pubkey)
   const hasActivity = checkAccountHasActivity(account)
 
-  return { accountId, hasActivity }
+  return { accountId, accountMetadata, hasActivity }
 }
 
 export const accountManagement = createQueryKeys('accountManagement', {
-  accountIdWithActivity: (accountNumber: number, chainId: ChainId, wallet: HDWallet) => ({
-    queryKey: ['accountIdWithActivity', chainId, wallet.getDeviceID(), accountNumber],
+  accountIdWithActivityAndMetadata: (
+    accountNumber: number,
+    chainId: ChainId,
+    wallet: HDWallet,
+    walletDeviceId: string,
+  ) => ({
+    queryKey: ['accountIdWithActivityAndMetadata', chainId, walletDeviceId, accountNumber],
     queryFn: () => {
-      return getAccountIdsWithActivity(accountNumber, chainId, wallet)
+      return getAccountIdsWithActivityAndMetadata(accountNumber, chainId, wallet)
     },
   }),
-  allAccountIdsWithActivity: (
+  allAccountIdsWithActivityAndMetadata: (
```

Comment:
Should we remove the `all` terminology from this to avoid any confusion?
It being *all* is really subjective, i.e is all, all the currently added ones (it isn't, but that's would be my assumption reading this)? All with certain heuristics (i.e can't fetch an infinity of numbers), in which case it isn't really *all*?

Reply:
sounds good, 831f5d87d79d60930f14a9d0b73ed6b9541944fb